Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/model/TaskViewModel.kt

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package org.dolphinemu.dolphinemu.model
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import kotlinx.coroutines.*
+
+class TaskViewModel : ViewModel() {
+    val result = MutableLiveData<Int>()
+    val isComplete = MutableLiveData<Boolean>()
+    var cancelled = false
+    var isRunning = MutableLiveData<Boolean>()
+
+    lateinit var resultGetter: () -> Unit
+    var onResultDismiss: (() -> Unit)? = null
+
+    init {
+        result.value = 0
+        isComplete.value = false
+        isRunning.value = false
```

Comment:
You could cut down a little on repeated code here by calling `clear()` if you add `isRunning.value = false` to `clear()`. (You can ignore this if you don't think it's a good suggestion.)

Reply:
Good idea.